[BUGIX] Do not register anonymous links

linawolf opened the pull request:
Anonymous links must not be registered as they have not name. Right now they are registered with their title.

This leads to multiple registrations of the same name eventually.

greg0ire commented on lib/Span/SpanProcessor.php:
```diff
@@ -167,66 +165,76 @@ private function replaceLinks(string $span): string
-            // we need to maintain the characters before and after the link
-            $prev = $match[1]; // previous character before the link
-            $next = $match[6]; // next character after the link
+    /** @param String[] $match */
```
```suggestion
    /** @param string[] $match */
```

greg0ire commented on lib/Span/SpanProcessor.php:
```diff
@@ -167,66 +165,76 @@ private function replaceLinks(string $span): string
+    }
 
-            $url = '';
+    /** @param String[] $match */
```
```suggestion
    /** @param string[] $match */
```

greg0ire reviewed (commented):
I suppose this registry we have in the environment serves some purpose for named links, but not for anonymous links, and never did, right? Does this really fix an issue (beyond using memory for nothing), or is it just a refactoring?

linawolf committed:
Update lib/Span/SpanProcessor.php

Co-authored-by: Grégoire Paris <postmaster@greg0ire.fr>

linawolf committed:
Update lib/Span/SpanProcessor.php

Co-authored-by: Grégoire Paris <postmaster@greg0ire.fr>

linawolf committed:
[BUGIX] Do not register anonymous links

Anonymous links must not be registered as they have not name. Right now they are registered with their title.

This leads to multiple registrations of the same name eventually.

[BUGIX] Do not register anonymous links

Anonymous links must not be registered as they have not name. Right now they are registered with their title.

This leads to multiple registrations of the same name eventually.

linawolf commented:
Right now if you would first register a  target with the anchor `name` and then an anonymous link with the title `name` the latter would override the first withouth this expecting to happen.

I noted this while working on warnings for duplicate registered link targets. All the anonymous links were popping up as unexpected warnings.

SenseException commented on tests/SpanProcessorTest.php:
```diff
@@ -0,0 +1,35 @@
+
+    public function testProcessingLinkSpanSetsLinkTarget(): void
+    {
+        $this->environment->expects(self::exactly(1))->method('setLinkTarget');
+        $spanProcessor = new SpanProcessor($this->environment, '`example <http://example.org/>`_');
```
I'd feel more secure about that test when `setLinkTarget` confirms that it gets the expected value.

linawolf replied:
setLinkTarget gets called with an object of LinkTarget. I didn't find a way to test that and thought it is covered in the Functional tests then. Any idea?

linawolf replied:
@SenseException , maybe like this with a callback?

SenseException replied:
I think a simple instance creation of the object with the same expected constructor values should do the trick too. AFAIK objects are compared with `==` in `with()`
https://3v4l.org/GBLYU

linawolf replied:
Cool, didn't know that

linawolf committed:
[Feature] Register link anchors globally in Metas

Introduce callback tests

SenseException commented:
Thank you @linawolf
